sql: support SHOW DATABASES WITH COMMENTS

hueypark opened the pull request:
Informs #36439.

Release note (sql change): support SHOW DATABASES WITH COMMENTS
now supports printing out database comments using the optional phrase
`WITH COMMENT`, e.g `SHOW DATABASES WITH COMMENT`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/37175)
<!-- Reviewable:end -->

jordanlewis reviewed (approved):
LGTM, thank you @hueypark!

jordanlewis commented:
bors r+

hueypark committed:
sql: support SHOW DATABASES WITH COMMENTS

Release note (sql change): support SHOW DATABASES WITH COMMENTS
now supports printing out database comments using the optional phrase
`WITH COMMENT`, e.g `SHOW DATABASES WITH COMMENT`.

hueypark commented:
@jordanlewis I fixed the test failure. Please check again.

jordanlewis commented:
bors r+

Thanks again!
